.github/workflows/zenodo-cache-sync.yml

```diff
@@ -41,8 +41,6 @@ jobs:
 
       - name: Checkout desired branch
         uses: actions/checkout@v4
-        with:
-          ref: ${{ env.GITHUB_REF }}
```

Comment:
This deletion was an effort to fix test failures resulting from the fact that this PR is from a forked repo. But those tests still fail due to permissions anyways, so idk whether it's worth keeping this out or not. @bendnorman @jdangerx do you have thoughts?

Reply:
@jdangerx any idea why `pytest.yml` uses `fetch-depth: 2` instead of the default `fetch-depth: 1`?

Reply:
I bet there's something somewhere in our tox-pytest that cares about "what has changed since the previous commit." Not sure what that would be, though.

Reply:
I think it's worth keeping it out, but point taken that the GCS credentials are obviously not going to be available in fork PRs. If we get more time to think about contributor workflows in the future we should think a bit harder about how we want our CI checks to go.

Reply:
In the `update-conda-lockfile` action there's a check for one of the actions that it's not on a forked repo. We could add that where appropriate to skip things that should run from outside the org.